builder/azure/common/client/config.go

```diff
@@ -126,7 +126,7 @@ func (c *Config) setCloudEnvironment() error {
 
 	environmentContext, cancel := context.WithTimeout(context.Background(), time.Minute*3)
 	defer cancel()
-	env, err := environments.FromEndpoint(environmentContext, c.MetadataHost, c.CloudEnvironmentName)
```

Comment:
CloudEnvironmentName was removed from the `FromEndpoint` function in a recent update to go-azure-sdk, this was used as a fall back mechanism if metadata host is not set, we default set that to Azure Public, but below on line 132 we fall back to calling the FromName method to get the environment, so this should not be a breaking change.